pyk/src/pyk/kast/utils.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, cast
+
+from ..konvert import _ast_to_kast
+from .markdown import select_code_blocks
+from .outer_parser import OuterParser
+from .outer_syntax import Definition
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+    from pathlib import Path
+    from typing import Final
+
+    from .outer import KDefinition
+    from .outer_syntax import Module
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+def _slurp(
+    definition_text: str,
+    search_paths: Iterable[Path] = (),
+    processed_files: Iterable[Path] = (),
+    md_selector: str = 'k',
+) -> tuple[Module, ...]:
+    parser = OuterParser(definition_text)
+    definition = parser.definition()
+    result = definition.modules
+    for require in definition.requires:
+        try_files = [include_dir / require.path for include_dir in search_paths]
+        try:
+            # Get the first source file we can find by iterating through search_paths
+            index = [file.exists() for file in try_files].index(True)
+        except ValueError:
+            _LOGGER.critical(f'{require.path} not found')  # TODO Print the source location of the requires clause
+            _LOGGER.critical(f'Lookup directories: {[str(path) for path in search_paths]}')
+            exit(1)
```

Comment:
`exit` should not be called here since we might want to call `parse_outer` from contexts other than the CLI. Instead, consider re-raising the exception and catching it in `__main__` to do the `exit`.